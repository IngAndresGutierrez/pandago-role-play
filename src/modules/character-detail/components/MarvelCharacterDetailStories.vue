<template>
  <v-container class="py-4">
    <v-row justify="center">
      <v-col cols="12">
        <v-carousel v-if="characterStories.length" height="auto" hide-delimiters>
          <v-carousel-item v-for="(stories, index) in characterStories" :key="index">
            <v-card-title v-if="stories?.title">{{ stories.title }}</v-card-title>
            <v-card-text v-if="stories?.description">{{ stories.description }}</v-card-text>

            <v-card-title v-if="stories?.creators?.items?.length" class="title">Creadores</v-card-title>

            <v-list v-if="stories?.creators?.items?.length" lines="two">
              <v-list-item
                v-for="creator in stories.creators.items"
                :key="creator.name"
                :subtitle="creator.role"
                :title="creator.name"
              />
            </v-list>
          </v-carousel-item>
        </v-carousel>
        <v-container v-if="loading">
          <v-row justify="center">
            <v-progress-circular color="primary" indeterminate />
          </v-row>
        </v-container>
        <v-alert v-if="!characterStories.length && !loading" type="info">Este personaje no aparece en ninguna
          historia.</v-alert>
      </v-col>
    </v-row>
  </v-container>
</template>

<script setup lang="ts">
  import { onMounted } from 'vue'
  import { useRoute } from 'vue-router'

  import marvelService from '../services/character'

  const route = useRoute()
  const characterStories = ref(null)
  const loading = ref(false)

  const fetchCharacterDetailStories = async () => {
    loading.value = true

    if (route.params.id) {
      const characterId = Number(route.params.id)
      characterStories.value = await marvelService.getCharacterDetailStories(characterId)
    }

    loading.value = false
  }

  onMounted(fetchCharacterDetailStories)

</script>

<style scoped>
.title {
  text-align: center;
}
</style>
