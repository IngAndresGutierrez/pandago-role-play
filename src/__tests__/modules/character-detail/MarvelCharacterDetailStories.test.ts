import 'vuetify/styles'
import { mount } from '@vue/test-utils'
import { createVuetify } from 'vuetify'
import { describe, expect, test } from 'vitest'

import MarvelCharacterDetailStories from '@/modules/character-detail/components/MarvelCharacterDetailStories.vue'

const vuetify = createVuetify()

const mockCharacter = {
  id: 1,
  name: 'Test character',
  thumbnail: {
    path: 'https://mypath.com/images',
    extension: 'jpg',
  },
}

describe('MarvelCharacterDetailStories.vue', () => {
  test('renders correctly', () => {
    const wrapper = mount(MarvelCharacterDetailStories, {
      global: {
        plugins: [vuetify],
      },
      props: {
        character: mockCharacter,
      },
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
