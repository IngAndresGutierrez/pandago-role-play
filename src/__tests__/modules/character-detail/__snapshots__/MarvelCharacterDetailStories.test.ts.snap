// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MarvelCharacterDetailStories.vue > renders correctly 1`] = `
<div
  character="[object Object]"
  class="v-container v-locale--is-ltr py-4"
  data-v-971431d5=""
>
  <div
    class="v-row justify-center"
    data-v-971431d5=""
  >
    <div
      class="v-col v-col-12"
      data-v-971431d5=""
    >
      <!--v-if-->
      <!--v-if-->
      <div
        class="v-alert v-theme--light bg-info v-alert--density-default v-alert--variant-flat"
        data-v-971431d5=""
        role="alert"
      >
        
        <!---->
        <span
          class="v-alert__underlay"
        />
        
        <!---->
        <div
          class="v-alert__prepend"
        >
          <i
            aria-hidden="true"
            class="mdi-information mdi v-icon notranslate v-theme--light"
            density="default"
            style="font-size: 28px; height: 28px; width: 28px;"
          />
        </div>
        <div
          class="v-alert__content"
        >
          <!---->
          <!---->
          
          Este personaje no aparece en ninguna historia.
          
        </div>
        <!---->
        <!---->
      </div>
    </div>
  </div>
</div>
`;
